backend/src/main/java/ai/verta/modeldb/experimentRun/ExperimentRunDAORdbImpl.java

```diff
@@ -1216,7 +1218,14 @@ public List<String> getAccessibleExperimentRunIDs(
           "Access is denied. ExperimentRun not found for given ids : " + requestedExperimentRunIds);
     }
     Set<String> projectIdSet = new HashSet<>(projectIdExperimentRunIdMap.values());
-    allowedProjectIds.retainAll(projectIdSet);
+    boolean allowedAllResources = RoleServiceUtils.checkAllResourceAllowed(allowedProjects);
+    Set<String> allowedProjectIds;
+    if (allowedAllResources) {
+      allowedProjectIds = projectIdSet;
+    } else {
+      allowedProjectIds = RoleServiceUtils.getResourceIds(allowedProjects);
+      allowedProjectIds.retainAll(projectIdSet);
+    }
```

Comment:
this same block of code seems to be repeated. Worth making it a method in RoleServiceUtils so it's not repeated everywhere?

Reply:
yes, extracted